wagon-providers/wagon-http-shared/src/main/java/org/apache/maven/wagon/shared/http/EncodingUtil.java

```diff
@@ -77,29 +83,28 @@ public static String encodeURLToString( String url )
     /**
      * Parses and returns an encoded version of the given URL string alongside the given paths.
      *
-     * @param baseUrl Base URL to use when constructing the final URL, ie: scheme://authority/initial.path.
-     * @param paths   Additional path(s) to append at the end of the base path.
+     * @param baseUrl Base URL to use when constructing the final URL. This has to be a valid URL already.
+     * @param paths   Additional unencoded path(s) to append at the end of the base path.
      * @return Composed URL (base + paths) already encoded, separating the individual path components by "/".
      * @since TODO
      */
     public static String encodeURLToString( String baseUrl, String... paths )
     {
         StringBuilder url = new StringBuilder( baseUrl );
 
-        String[] parts = paths == null ? //
+        String[] segments = paths == null ? //
             new String[0] : //
             paths.length == 1 ? paths[0].split( "/" ) : paths;
```

Comment:
@michael-o This part looks a bit dodgy to me. I would rather use two distinct methods here `encodeURLToString( String baseUrl, String... pathSegments )` and `encodeURLToString( String baseUrl, String path )` with the latter splitting the path into segments and then calling the former. Otherwise looks OK to me.

Reply:
This might be problematic because how should the compiler know `#encodingURLToString("../dd", "path/d/c")` should not be passed to the varargs overload?

Reply:
@michael-o Different (more descriptive) method names?

Reply:
I fiddled with it a bit on Zulu 7, Oracle JDK 7 + 11 and the overload is properly inferred. There is only one edge case. When you have the path *segment* `a/b` and you pass it, the single string overload will be used. You have to wrap into an array.

Reply:
As soon as Jenkins build passes, I will merge into master. Thank you again for the advise.